### PR TITLE
Sync with Make setters and getters optional

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #372

# Description

Removes setters and getters from contribs.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [X] Yes. - Link to PR:  https://github.com/open-telemetry/opentelemetry-python/pull/1690
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def extract(
 
     def inject(
         self,
-        set_in_carrier: Setter[TextMapPropagatorT],
+        setter: Setter[TextMapPropagatorT],
```
Comment (codeboten) [inline propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py:108]: should this parameter be optional here? it also appears to be in the wrong order

File: sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py [hunk 2]
```diff
@@ -267,7 +267,7 @@ def _parse_sampled_flag(sampled_flag_str):
 
     def inject(
         self,
-        set_in_carrier: Setter[TextMapPropagatorT],
+        setter: Setter[TextMapPropagatorT],
```
Comment (codeboten) [inline sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py:270]: The setter here also appears to be in the wrong order

File: sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py [hunk 3]
```diff
@@ -395,13 +380,13 @@ def test_fields(self, mock_trace):
             }
         )
 
-        mock_set_in_carrier = Mock()
+        mock_setter = Mock()
 
-        AwsXRayPropagatorTest.XRAY_PROPAGATOR.inject(mock_set_in_carrier, {})
+        AwsXRayPropagatorTest.XRAY_PROPAGATOR.inject(mock_setter, {})
```
Comment (codeboten) [inline sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py:385]: setter should be an optional parameter

File: propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py (hunk 1 above)
Comment (ocelotl) [reply propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py]: fixed

File: sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py (hunk 3 above)
Comment (ocelotl) [reply sdk-extension/opentelemetry-sdk-extension-aws/tests/trace/propagation/test_aws_xray_format.py]: fixed

File: sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py (hunk 2 above)
Comment (ocelotl) [reply sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py]: fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python-contrib/compare/v0.18b0...HEAD)
+- Sync contribs with optional getters and setters
```
Comment (codeboten) [inline CHANGELOG.md:8]:
```suggestion
- Make getters and setters optional
```

**Review by codeboten — APPROVED** [review]
Minor nit on the changelog entry, otherwise looks good!

**Review by owais — APPROVED** [review]
Looks good. The API definitely feels nicer. Thanks @ocelotl

**Pull request merged**